src/filters/truncate.js

```diff
@@ -3,7 +3,7 @@
 angular.module('openshiftCommonUI')
   .filter('truncate', function() {
     return function(str, charLimit, useWordBoundary, newlineLimit) {
-      if (!str) {
+      if (typeof str !== 'string' || str.length <= charLimit) {
```

Comment:
My complaint is that it changes the behavior of `"foo\nbar\n"` where length limit is 80 and newline limit is 1. Before this would remove the second line and now it will return the string as-is. I think we can fix the immediate bug while still keeping the newline limit without too many other changes.

But I think the problem might be different here: If truncate is 80 characters and the string is 81, we're taking off one character and appending and ellipsis, which actually makes the string longer. Or is there another issue at play?